components/ActivityForm/index.js

```diff
@@ -2,6 +2,123 @@ import styled from "styled-components";
 import { useRouter } from "next/router";
 import categories from "@/assets/categories";
 import { useState } from "react";
+import Link from "next/link";
+import { StyledButton } from "../ActivityDetails";
+
+export default function ActivityForm({ activity, onSubmit, isUpdateMode }) {
+  const [selectedCategoryIds, setSelectedCategoryIds] = useState(
+    activity?.categoryIds || []
+  );
+
+  const router = useRouter();
+
+  function handleChange(event) {
```

Comment:
```suggestion
  function handleChangeCategory(event) {
```
The name is absolutly fine already, my suggestion would just be to make it a bit more specific because only fires on category changes. E.g. `handleCategoryChange` or `handleChangeCategory`